core/src/test/java/org/infinispan/executors/LimitedExecutorTest.java

```diff
@@ -57,11 +57,13 @@ public void testConcurrencyLimit() throws Exception {
       limitedExecutor.execute(() -> cf2.complete("bla"));
 
       Thread.sleep(10);
+      assertFalse(cf1.isDone());
       assertFalse(cf2.isDone());
 
       latch.countDown();
       assertEquals("bla", cf1.get(10, SECONDS));
       assertEquals("bla", cf2.get(10, SECONDS));
+      eventuallyEquals(0, executor::getActiveCount);
```

Comment:
Okay, so the 2 threads are there to allow the possibility that the tasks will run in parallel, right? But what do you actually test here? The test confirms that eventually both tasks will be executed, but there's nothing testing the level of parallelism.